docs/developers.md

```diff
@@ -74,3 +74,61 @@ Read more:
 
 - Effective Go [introduces formatting](https://golang.org/doc/effective_go.html#formatting).
 - The Go Wiki has a great article on [formatting](https://github.com/golang/go/wiki/CodeReviewComments).
+
+### Unit Tests
+
+Unit tests may be run locally via `make test`.
+
+### End-to-end Tests
+
+End-to-end tests for the DCOS, Kubernetes and OpenShift orchestrators may be run
+via `make test-{dcos,kubernetes,openshift}`.  The test process can optionally
+deploy and tear down a cluster as part of the test (this is enabled by default).
+You'll need access to an Azure subscription, as well as at least the following
+environment variables to be set:
+
+* `ACCOUNT_SUBSCRIPTION_ID`: Azure subscription UUID
+* `ACCOUNT_TENANT_ID`: Azure tenant UUID
+* `ACCOUNT_USER_CLIENT_ID`: Azure client ID
+* `ACCOUNT_USER_CLIENT_SECRET`: Azure client secret
+
+* `PROVISIONER_ACCOUNT_SUBSCRIPTION_ID`: same as ACCOUNT_SUBSCRIPTION_ID
+* `PROVISIONER_ACCOUNT_TENANT_ID`: same as ACCOUNT_TENANT_ID
+* `PROVISIONER_ACCOUNT_USER_CLIENT_ID`: same as ACCOUNT_USER_CLIENT_ID
+* `PROVISIONER_ACCOUNT_USER_CLIENT_SECRET`: same as ACCOUNT_USER_CLIENT_SECRET
+
+* `USER_CLIENT_ID`: same as ACCOUNT_USER_CLIENT_ID
+* `USER_CLIENT_SECRET`: same as ACCOUNT_USER_CLIENT_SECRET
+
+#### OpenShift
+
+To test the OpenShift orchestrator, you'll need to enable programmatic
+deployment of the underlying image.  In the Azure console, find the image under
+Home > New > Marketplace > Everything.  Click "Want to deploy programmatically?
+Get started".  Enable your subscription and click Save.
+
+You'll also need to have `oc` and `kubectl` binaries locally in your PATH which
+correspond to the cluster version being tested.  Download the `oc` binary, then
+make a symlink or copy of it and name the new file `kubectl`.
+
+To have the test process deploy and tear down a cluster, set the following
+environment variables:
+
+* `CLUSTER_DEFINITION=examples/openshift.json`
+* `DISTRO=openshift39_centos`
+* `LOCATION=eastus`
+* `SUBSCRIPTION_ID`: same as ACCOUNT_SUBSCRIPTION_ID
+* `TENANT_ID`: same as ACCOUNT_TENANT_ID
+
+Alternatively, to run tests on a pre-deployed OpenShift cluster, set the
+following environment variables:
+
+* `CLEANUP_ON_EXIT=false`
+* `CLEANUP_IF_FAIL=false`
+* `LOCATION=eastus`
```

Comment:
Is location always `eastus`?

Reply:
It is until we ship reliable images across multiple locations.